addons/base_vat/models/res_partner.py

```diff
@@ -77,14 +77,18 @@
 
 _region_specific_vat_codes = {
     'xi',
+    't',
 }
 
 
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
     def _split_vat(self, vat):
-        vat_country, vat_number = vat[:2].lower(), vat[2:].replace(' ', '')
+        if vat[1].isalpha():
+            vat_country, vat_number = vat[:2].lower(), vat[2:].replace(' ', '')
+        else:
+            vat_country, vat_number = vat[:1].lower(), vat[1:].replace(' ', '')
```

Comment:
Maybe touching this is not really what we want for a specific case like this task, i'm currently working on a similar task i will try to make it more loca dependant :+1:, we will also see what Quentin thinks about this :smile:

Reply:
Tell me if i'm wrong but we could do a method check_vat_jp that receive the vat being in either JP.... or T.... format and return true if the format are good and false otherwise. 
From my understanding of the thing i will go through the methode a first time with only the numbers and it will return false since it's not starting with JP or T, then it will call again check_vat_jp with the good vat number (JP.... or T....) and we only have to check with some if and elif if the format is correct.

Reply:
i've made this pr for hungary https://github.com/odoo/odoo/pull/136744#issuecomment-1736789165 where the vat number must either be HU.... or a specific format of number and it seems to be working fine

Reply:
I might be wrong but I think that the "check_vat_jp" would not trigger when entering a vat number starting with "T". These methods are called by a gettattr which concatenates "check_vat_" with the country code entered at the beginning of the vat number so a code starting with "T" always calls "check_vat_t". 

Maybe in your case, it worked because the second type of vat number doe not have a country code and in these cases, the system calls the method with the country code of the country of the company but it wouldn't work if there was a country code DIFFERENT from HU at its beginning. Which is the specific scenario I have to handle.

Again, might be wrong...

Reply:
Yeah i see :thinking:, Let's see what Quentin thinks when he review the pr :smile: